store/db.go

```diff
@@ -44,39 +48,49 @@ func NewDB(dbdir string) (*DB, error) {
 
 func (db *DB) Open() error {
 	// take a lock on db dir
-	if db.lock != nil {
-		panic("cas db lock already gained")
-	}
 	dl, err := lock.ExclusiveLock(db.dbdir, lock.Dir)
 	if err != nil {
 		return err
 	}
-	db.lock = dl
 
 	sqldb, err := sql.Open("ql", filepath.Join(db.dbdir, DbFilename))
 	if err != nil {
 		dl.Close()
 		return err
 	}
+
+	// (XXX): flock is already granted, the mutex
```

Comment:
@jonboulle I can't agree, see man 2 flock:

> If  a  process  uses  open(2) (or similar) to obtain more than one descriptor for the same file, these descriptors are treated independently by flock().  An attempt to lock the file using one of these
       file descriptors may be denied by a lock that the calling process has already placed via another descriptor.